INCWaveMountainsView/Classes/INCWaveMountainsView.m

```diff
@@ -0,0 +1,591 @@
+        NSArray *colors = @[(id)[UIColor colorWithRed:48.0/255.0 green:35.0/255.0 blue:174.0/255.0 alpha:0.4].CGColor, (id)[UIColor colorWithRed:180.0/255.0 green:236.0/255.0 blue:81.0/255.0 alpha:0.4].CGColor];
+        _gradientCenterMountain.colors = colors;
+    }
+    return _gradientCenterMountain;
+}
+
+-(CAGradientLayer *)gradientRightMountain
+{
+    if (!_gradientRightMountain) {
+        _gradientRightMountain = [CAGradientLayer layer];
+        
+        _gradientRightMountain.startPoint = CGPointMake(0, 0);
+        _gradientRightMountain.endPoint = CGPointMake(1, 0);
+        NSArray *colors = @[(id)[UIColor colorWithRed:48.0/255.0 green:35.0/255.0 blue:174.0/255.0 alpha:0.4].CGColor, (id)[UIColor colorWithRed:180.0/255.0 green:236.0/255.0 blue:81.0/255.0 alpha:0.4].CGColor];
+        _gradientRightMountain.colors = colors;
+    }
+    return _gradientRightMountain;
+}
+
+-(CAShapeLayer *)maskLeftMountain
+{
+    if (!_maskLeftMountain) {
+        _maskLeftMountain = [CAShapeLayer layer];
+        _maskLeftMountain.fillColor = [UIColor whiteColor].CGColor;
+    }
+    return _maskLeftMountain;
+}
+
+-(CAShapeLayer *)maskCenterMountain
+{
+    if (!_maskCenterMountain) {
+        _maskCenterMountain = [CAShapeLayer layer];
+        _maskCenterMountain.fillColor = [UIColor whiteColor].CGColor;
+    }
+    return _maskCenterMountain;
+}
+
+-(CAShapeLayer *)maskRightMountain
+{
+    if (!_maskRightMountain) {
+        _maskRightMountain = [CAShapeLayer layer];
+        _maskRightMountain.fillColor = [UIColor whiteColor].CGColor;
+    }
+    return _maskRightMountain;
+}
+
+-(CAShapeLayer *)shapeLayerLeftMountain
+{
+    if (!_shapeLayerLeftMountain) {
+        _shapeLayerLeftMountain = [[CAShapeLayer alloc]init];
+#pragma mark pending to create external colours for change the mountains
+        [_shapeLayerLeftMountain setFillColor:[UIColor clearColor].CGColor];
+        [_shapeLayerLeftMountain addSublayer:self.gradientLeftMountain];
+
+        [self.layer addSublayer:_shapeLayerLeftMountain];
+    }
+    return _shapeLayerLeftMountain;
+}
+
+-(CAShapeLayer *)shapeLayerCenterMountain
+{
+    if (!_shapeLayerCenterMountain) {
+        _shapeLayerCenterMountain = [[CAShapeLayer alloc]init];
+        
+        [_shapeLayerCenterMountain setFillColor:[UIColor clearColor].CGColor];
+        [_shapeLayerCenterMountain addSublayer:self.gradientCenterMountain];
+
+        [self.layer addSublayer:_shapeLayerCenterMountain];
+    }
+    return _shapeLayerCenterMountain;
+}
+
+-(CAShapeLayer *)shapeLayerRightMountain
+{
+    if (!_shapeLayerRightMountain) {
+        _shapeLayerRightMountain = [[CAShapeLayer alloc]init];
+        
+        [_shapeLayerRightMountain setFillColor:[UIColor clearColor].CGColor];
+        [_shapeLayerRightMountain addSublayer:self.gradientRightMountain];
+
+        [self.layer addSublayer:_shapeLayerRightMountain];
+    }
+    return _shapeLayerRightMountain;
+}
```

Comment:
Could each Mountain be a class, so you don't have to repeat the same code for each one?